syntax = "proto3";
option go_package = "gen/common";

message SearchRequest {
	repeated float vec = 1;
	int32 k = 2;
}

message SearchResponse {
	int32 code = 1;
	string msg = 2;
	repeated float distance = 3;
	repeated int64 labels = 4;
}

service FaissIndex {

	rpc Search(SearchRequest) returns (SearchResponse) {}
}